scipy/optimize/tests/test_nonlin.py

```diff
@@ -34,6 +34,8 @@ def F(x):
 
 F.xin = [1,1,1,1,1]
 F.KNOWN_BAD = {}
+F.JAC_KSP_BAD = {}
+F.ROOT_JAC_KSP_BAD = {}
```

Comment:
There's quite a few solvers which end up in `JAC_KSP_BAD` for one or more problems. Maybe it's a good idea to add a description here about why, and how much of a problem it is? Or open an issue and refer to it in a comment here, if it needs solving?

Reply:
I add some comments in new commit